examples/ffi/exporter.cpp

```diff
@@ -40,8 +40,9 @@ int main(int argc, char *argv[]) {
 
   const ddog_prof_Slice_ValueType sample_types = {&wall_time, 1};
   const ddog_prof_Period period = {wall_time, 60};
-  std::unique_ptr<ddog_prof_Profile, Deleter> profile{
+  ddog_prof_Profile profile_data{
       ddog_prof_Profile_new(sample_types, &period, nullptr)};
+  std::unique_ptr<ddog_prof_Profile, Deleter> profile{&profile_data};
```

Comment:
Minor: I wonder if such a simple example needs this kind of sophistication, e.g. maybe we could simplify this a bit?

Reply:
I just fixed it up for the changes I made ^_^

I get that in C++, they want to have destructors run, rather than doing it manually. But obviously the C bindings don't have destructors.

I think we can maybe simplify this code, or acknowledge that we care more about C++ and try to _also_ generate C++ headers and bindings, or something.